grpc.go

```diff
@@ -415,16 +415,21 @@ func (a *agentGRPC) postExecProcess(ctr *container, proc *process) error {
 // path set by the spec, since we will always ignore it. Indeed, it makes no
 // sense to rely on the namespace path provided by the host since namespaces
 // are different inside the guest.
-func (a *agentGRPC) updateContainerConfigNamespaces(config *configs.Config) error {
+func (a *agentGRPC) updateContainerConfigNamespaces(config *configs.Config, ctr *container) {
 	var pidNs, ipcNs, utsNs bool
 
 	// Update shared PID namespace.
 	for idx, ns := range config.Namespaces {
 		if ns.Type == configs.NEWPID {
-			// In case the path is empty because we don't expect
-			// the containers to share the same PID namespace, a
-			// new PID ns is going to be created.
-			config.Namespaces[idx].Path = a.sandbox.sharedPidNs.path
+			// Use shared pid ns if SandboxPidns has been set in either
+			// the CreateSandbox request or CreateContainer request.
+			// Else set this to empty string so that a new pid namespace is
+			// created for the container.
+			if ctr.useSandboxPidNs || a.sandbox.sandboxPidNs {
```

Comment:
hmm, so `a.sandbox.sandboxPidNs` still means all containers will join the kata agent pause process held sandbox level pidns. Is it intentional? I don't see any issue with it but want to make sure we are on the same page about its semantics. And whatever we decide, we need to document it in the proto file.